cpp/Platform.Setters/Setter[TResult, TDecision].h

```diff
@@ -1,46 +1,47 @@
-﻿namespace Platform::Setters
-{
-    template <typename ...> class Setter;
-    template <typename TResult, typename TDecision> class Setter<TResult, TDecision> : public SetterBase<TResult>
-    {
-        private: TDecision _trueValue = 0;
-        private: TDecision _falseValue = 0;
-
-        public: Setter(TDecision trueValue, TDecision falseValue, TResult defaultValue)
-            : SetterBase(defaultValue)
-        {
-            _trueValue = trueValue;
-            _falseValue = falseValue;
-        }
-
-        public: Setter(TDecision trueValue, TDecision falseValue) : this(trueValue, falseValue, 0) { }
-
-        public: Setter(TResult defaultValue) : SetterBase(defaultValue) { }
-
-        public: Setter() { }
-
-        public: TDecision SetAndReturnTrue(TResult value)
-        {
-            _result = value;
-            return _trueValue;
-        }
-
-        public: TDecision SetAndReturnFalse(TResult value)
-        {
-            _result = value;
-            return _falseValue;
-        }
-
-        public: TDecision SetFirstAndReturnTrue(IList<TResult> &list)
-        {
-            _result = list[0];
-            return _trueValue;
-        }
-
-        public: TDecision SetFirstAndReturnFalse(IList<TResult> &list)
-        {
-            _result = list[0];
-            return _falseValue;
-        }
-    };
-}
+﻿namespace Platform::Setters
+{
+    template <typename ...> class Setter;
+    template <typename TResult, typename TDecision> class Setter<TResult, TDecision> : public SetterBase<TResult>
+    {
+        private: TDecision _trueValue = 0;
+        private: TDecision _falseValue = 0;
+
+        using base = SetterBase<TResult>;
+
+        public: Setter(TDecision trueValue, TDecision falseValue, TResult defaultValue)
+        {
+            _trueValue = trueValue;
+            _falseValue = falseValue;
+            base::_result = std::move(defaultValue);
+        }
+
+        public: Setter(TDecision trueValue, TDecision falseValue) : Setter(trueValue, falseValue, 0) { }

```

Comment:
If `TDecision` might be big size type, use `std::move`

Reply:
No, usually it is integral type.

Reply:
@Mitron57 use default values instead of zeros where it is possible. At the moment we use integral types, but if we need we could use non-integral in the future. Default values will make experiments possible.